python/src/model.rs

```diff
@@ -1115,18 +1116,16 @@ impl<'a> From<&'a PyNamedOrBlankNodeRef<'a>> for NamedOrBlankNodeRef<'a> {
     }
 }
 
-impl<'a> TryFrom<&'a PyAny> for PyNamedOrBlankNodeRef<'a> {
-    type Error = PyErr;
-
-    fn try_from(value: &'a PyAny) -> PyResult<Self> {
-        if let Ok(node) = value.extract::<PyRef<'_, PyNamedNode>>() {
+impl<'py> FromPyObject<'py> for PyNamedOrBlankNodeRef<'py> {
+    fn extract_bound(ob: &Bound<'py, PyAny>) -> PyResult<Self> {
+        if let Ok(node) = ob.extract::<PyRef<'_, PyNamedNode>>() {
```

Comment:
Are the type ascriptions necessary here?

Reply:
Not really but it allows to avoid [a false positive of `clippy::same-functions-in-if-condition`](https://github.com/rust-lang/rust-clippy/issues/10928)